modules/javafx.graphics/src/main/java/com/sun/javafx/image/impl/BaseIndexedToByteConverter.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javafx.image.impl;
+
+import com.sun.javafx.image.AlphaType;
+import com.sun.javafx.image.BytePixelGetter;
+import com.sun.javafx.image.BytePixelSetter;
+import com.sun.javafx.image.IndexedToBytePixelConverter;
+import com.sun.javafx.image.PixelUtils;
+import java.nio.ByteBuffer;
+
+abstract class BaseIndexedToByteConverter implements IndexedToBytePixelConverter {
+
+    static abstract class IndexedGetter implements BytePixelGetter {
+        final int[] preColors;
+        final int[] nonPreColors;
+        private final Boolean premultiplied;
+
+        IndexedGetter(int[] colors, Boolean premultiplied) {
```

Comment:
I think using `Boolean` to store 3 values is confusing.  Use an `enum`?

Reply:
Replaced with `AlphaType`.